File: app/src/main/java/com/example/dota2herodhevantharezamaap/HeroAdapter.java
package com.example.dota2herodhevantharezamaap;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;

import java.util.ArrayList;

public class HeroAdapter extends RecyclerView.Adapter<HeroAdapter.CardViewHolder> {
    private ArrayList<Hero> listHero;
    public HeroAdapter(ArrayList<Hero> list) {
        this.listHero = list;
    }
    @NonNull
    @Override
    public CardViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_cardview, viewGroup, false);
        return new CardViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final CardViewHolder holder, int position) {
        final Hero hero = listHero.get(position);
        Glide.with(holder.itemView.getContext())
                .load(hero.getImage())
                .apply(new RequestOptions().override(350,550))
                .into(holder.imgPhoto);
        holder.tvName.setText(hero.getName().toUpperCase());
        holder.btnDetail.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(holder.itemView.getContext(), DetailActivity.class);
                intent.putExtra(DetailActivity.EXTRA_NAME, hero.getName());
                intent.putExtra(DetailActivity.EXTRA_DETAIL, hero.getDetail());
                intent.putExtra(DetailActivity.EXTRA_IMAGE, hero.getImage());
                holder.itemView.getContext().startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return listHero.size();
    }

    public class CardViewHolder extends RecyclerView.ViewHolder {
        ImageView imgPhoto;
        TextView tvName;
        Button btnDetail;
        CardViewHolder(@NonNull View itemView) {
            super(itemView);
            imgPhoto = itemView.findViewById(R.id.img_item_photo);
            tvName = itemView.findViewById(R.id.tv_item_name);
            btnDetail = itemView.findViewById(R.id.btn_detail);
        }
    }
}

File: app/src/main/java/com/example/dota2herodhevantharezamaap/kotlin/adapter/HeroAdapter.kt
package com.example.dota2herodhevantharezamaap.kotlin.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.example.dota2herodhevantharezamaap.R
import com.example.dota2herodhevantharezamaap.kotlin.activity.DetailActivity
import com.example.dota2herodhevantharezamaap.kotlin.model.Hero
import kotlinx.android.synthetic.main.item_cardview.view.*
import org.jetbrains.anko.intentFor
import org.jetbrains.anko.sdk27.coroutines.onClick

class HeroAdapter(private val context: Context, private val list: ArrayList<Hero>) : RecyclerView.Adapter<HeroAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_cardview, parent, false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(list[position])
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind(hero: Hero) {
            Glide.with(itemView.context)
                    .load(hero.image)
                    .apply(RequestOptions().override(350, 550))
                    .into(itemView.img_item_photo)
            itemView.tv_item_name.text = hero.name
            itemView.onClick {
                itemView.context.startActivity(itemView.context.intentFor<DetailActivity>("HERO" to hero))
            }
            itemView.btn_detail.onClick {
                itemView.context.startActivity(itemView.context.intentFor<DetailActivity>("HERO" to hero))
            }
        }
    }

}